mlflow/store/db/utils.py

```diff
@@ -81,9 +82,12 @@ def make_managed_session():
         except MlflowException:
             session.rollback()
             raise
+        except sqlalchemy.exc.IntegrityError as e:
+            session.rollback()
+            raise MlflowException(message=str(e), error_code=RESOURCE_ALREADY_EXISTS)
```

Comment:
`RESOURCE_DOES_NOT_EXIST` may not be accurate for certain integrity errors, such as foreign key constraint violations. Can we use `BAD_REQUEST` instead?

```suggestion
            raise MlflowException(message=str(e), error_code=BAD_REQUEST)
```

Reply:
I think we should use RESOURCE_ALREADY_EXISTS, because we do converting here:
https://github.com/mlflow/mlflow/blob/56a733af68fde4c454d7ba193e65dec61bac649f/mlflow/server/handlers.py#L447
and the converter map is:
https://github.com/mlflow/mlflow/blob/56a733af68fde4c454d7ba193e65dec61bac649f/mlflow/exceptions.py#L23

so, RESOURCE_ALREADY_EXISTS will be converted into "400" code when return error code in http responding result.

Reply:
https://github.com/mlflow/mlflow/blob/352d6ae9c9adeff59a2a12bc7a0b35f18a992af7/mlflow/exceptions.py#L25

(If I'm not mistaken) `BAD_REQUEST` also will be converted to 400.

Reply:
oh. My link wrong. I am using "RESOURCE_ALREADY_EXISTS" in my code, not "RESOURCE_DOES_NOT_EXIST" .
"RESOURCE_ALREADY_EXISTS" should be more accurate than "BAD_REQUEST"

Reply:
```python
>>> MlflowException("foo", RESOURCE_ALREADY_EXISTS).get_http_status_code()
400
>>> MlflowException("foo", BAD_REQUEST).get_http_status_code()
400
```

Reply:
> "RESOURCE_ALREADY_EXISTS" should be more accurate than "BAD_REQUEST"

Can you tell us why?

Reply:
I rethink on this, `RESOURCE_ALREADY_EXISTS` might be confusing, the experiment is not a "resource", BAD_REQUEST code is better.